src/segmantic/seg/monai_unet.py

```diff
@@ -548,3 +695,104 @@ def print_table(header, vals, indent="\t"):
                 confusion_metrics,
                 (np.squeeze(x.cpu().numpy()) for x in conf_matrix.aggregate()),
             )
+
+
+def cross_validate(
+    image_dir: Path,
+    labels_dir: Path,
+    tissue_list: Path,
+    output_dir: Path,
+    config_files_dir: Path,
+    test_image_dir: Optional[Path] = None,
+    test_labels_dir: Optional[Path] = None,
+    n_splits: int = 7,
+    gpu_ids: List[int] = [0],
+):
+    print_config()
+    print("Cross-validating")
+    output_dir = Path(output_dir)
+    output_dir.mkdir(exist_ok=True)
+
+    tissue_dict = load_tissue_list(tissue_list)
+    print(tissue_dict)
+
+    data_dicts = PairedDataSet.create_data_dict(
+        image_dir=image_dir, labels_dir=labels_dir
+    )
+
+    test_data_dicts = []
+
+    if test_image_dir and test_labels_dir:
+        test_data_dicts = PairedDataSet.create_data_dict(
+            image_dir=test_image_dir, labels_dir=test_labels_dir
+        )
+
+    all_datafold_paths: List[Path] = PairedDataSet.kfold_crossval(
+        num_splits=n_splits,
+        data_dicts=data_dicts,
+        output_dir=output_dir,
+        test_data_dicts=test_data_dicts,
+    )
+
+    for config_file in Path(config_files_dir).iterdir():
+        # ToDo: add yaml file support
+        assert config_file.suffix in [".json"]
+
+        output_dir_scenario = output_dir / config_file.name
+        output_dir_scenario.mkdir(exist_ok=True)
+
+        for count, dataset_path in enumerate(all_datafold_paths):
+
+            current_output = output_dir_scenario / str(count)
+            print(current_output)
```

Comment:
tip: consider using the `logging` module for printing logs like this

Reply:
thats a nice idea~~ will have a look into it. Thank you